rust/xaynet-analytics/src/database/common.rs

```diff
@@ -111,3 +144,45 @@ where
         )
     }
 }
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct RelationalField {
+    pub value: String,
+    pub collection_name: String,
+}
+
+// NOTE: when split_once gets to stable, it would be a much better solution for this
+// https://doc.rust-lang.org/std/string/struct.String.html#method.split_once
+impl TryFrom<&str> for RelationalField {
+    type Error = anyhow::Error;
+
+    fn try_from(data: &str) -> Result<Self, Error> {
+        fn stringify(input: Option<&&str>) -> Result<String, Error> {
+            Ok(input
+                .ok_or_else(|| anyhow!("could not unwrap input"))?
+                .to_string())
+        }
+
+        let data_split: Vec<&str> = data.split('=').collect();
+        assert_eq!(data_split.len(), 2);
```

Comment:
Also here we can return an error instead of panicking.

Reply:
yup, makes sense